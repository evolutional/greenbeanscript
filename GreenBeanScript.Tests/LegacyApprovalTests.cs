using System;
using System.IO;
using System.Text;
using ApprovalTests;
using ApprovalTests.Reporters;
using GreenBeanScript.Libs;
using Microsoft.VisualStudio.TestTools.UnitTesting;

namespace GreenBeanScript.Tests
{
    [TestClass]
    [UseReporter(typeof (DiffReporter))]
    public class LegacyApprovalTests
    {
        private void Do(StringBuilder log, string libFileName)
        {
            Action<string> printFunc = (s) =>
            {
                log.AppendLine(s);
            };
            var stdLib = new StdLibrary(printFunc);
            var uut = new Machine(stdLib);
            var newLib = new Library();
            newLib.LoadFromFile(uut, libFileName);

            uut.ExecuteLibrary(newLib);
        }

        private void List(StringBuilder log, string libFileName)
        {
            var newLib = new Library();
            newLib.ListLibraryFromFile(log, libFileName);
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledHeapsort()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/heapsort.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledFib()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/fib.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledMatrix()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/matrix.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledHash()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/hash.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void CompiledFloat()
        {
            var log = new StringBuilder();
            Do(log, @"./ExampleScripts/float.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledFib()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/fib.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledMatrix()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/matrix.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledHeapsort()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/heapsort.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledHash()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/hash.gml");
            Approvals.Verify(log.ToString());
        }

        [UseReporter(typeof(DiffReporter))]
        [TestMethod]
        public void ListCompiledFloat()
        {
            var log = new StringBuilder();
            List(log, @"./../../ExampleScripts/float.gml");
            Approvals.Verify(log.ToString());
        }
    }
}
