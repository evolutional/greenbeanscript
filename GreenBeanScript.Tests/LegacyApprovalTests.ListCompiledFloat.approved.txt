GM LIBRARY C:\dev\greenbeanscript\GreenBeanScript.Tests\ExampleScripts\float.gml, DEBUG=0

STRING TABLE, SIZE=    9
===============================================================================
00000:IC
00003:print

NO SOURCE CODE IN LIB
FUNCTIONS, COUNT=1
===============================================================================

===============================================================================
FUNCTION ID 0
FLAGS 1
NUM PARAMS 0
NUM LOCALS 1
MAX STACK 3
BYTE CODE SIZE 68
  0000 push fp 29573.000000
  0008 set global 0
  0016 get global 0
  0024 set local 0
  0032 push null
  0036 get global 3
  0044 get global 0
  0052 call 1
  0060 pop
  0064 ret


END LIB
